lib/resty/iprepd.lua

```diff
@@ -0,0 +1,77 @@
+local cjson = require('cjson')
+local lrucache = require('resty.lrucache')
+local http = require('resty.http')
+
+function fatal_error(error_msg)
+  ngx.log(ngx.ERR, error_msg)
+  os.exit(1)
+end
+
+local _M = {}
+local mt = { __index = _M }
+
+function _M.new(options)
+  local iprepd_url = options.url or 'http://localhost:8080/'
+  local cache_ttl = options.cache_ttl or 30
+
+  local iprepd_threshold = options.threshold or fatal_error('Need to pass in a threshold')
+  local iprepd_api_key = options.api_key or fatal_error('Need to pass in an api_key')
+
+  -- TODO: Make configurable?
+  local cache, err = lrucache.new(200)
```

Comment:
Agreed that it would be nice if this was configurable just so we could experiment with increasing it if we ever wanted.